inst/htmlwidgets/plotly.js

```diff
@@ -62,10 +62,8 @@ HTMLWidgets.widget({
           attachKey("key");
           return obj; 
         });
-        Shiny.onInputChange(
-          ".clientValue-" + eventType + "-" + x.source, 
-          JSON.stringify(d)
-        );
+        var src = ".clientValue-" + eventType + "-" + x.source;
+        Shiny.onInputChange(src, JSON.stringify(d));
```

Comment:
Is there a reason you're doing `JSON.stringify()` and `jsonlite::fromJSON` explicitly? If you just pass `d` from here, and don't JSON-decode from R, I think it should work fine--we do the JSON encoding for you.